middleware/auth.go

```diff
@@ -0,0 +1,37 @@
+package middleware
+
+import (
+	"net/http"
+	"os"
+	"web/model"
+
+	"github.com/gin-contrib/sessions"
+	"github.com/gin-gonic/gin"
+)
+
+func Authentication() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		// before request
+		session := sessions.Default(c)
+		sessionID := session.Get("userID")
+
+		if sessionID == nil {
+			// Check for API key in the request
+			apiKey := c.GetHeader("X-API-Key")
+			if apiKey != os.Getenv("STATUSPAGE_API_KEY") {
+				c.AbortWithStatus(http.StatusUnauthorized)
+				return
+			}
+		} else {
+			userID := sessionID.(uint)
+			user, err := model.GetUserByID(userID)
```

Comment:
As this one looks very strange that you're getting the user from model, not the repository, which I wrote more about in other comment
https://github.com/LookinLabs/GoBlitz/pull/22/files#r1623394200